zmsentities/src/Zmsentities/Collection/AvailabilityList.php

```diff
@@ -246,4 +254,232 @@ public function withLessData(array $keepArray = [])
+                . "Neue Öffnungszeit:&thinsp;&thinsp;&thinsp;&thinsp;&thinsp;&thinsp;&thinsp;&thinsp;&thinsp;&thinsp;&thinsp;&thinsp;&thinsp;&thinsp;&thinsp;&thinsp;&thinsp;[$dateRange2, $timeRange2, Wochentag(e): " . $availability2->getWeekdayNames() . "]";
+        } else {
+            $message .= "Bestehende Öffnungszeit:&thinsp;&thinsp;[$dateRange2, $timeRange2, Wochentag(e): " . $availability2->getWeekdayNames() . "]\n"
+                . "Neue Öffnungszeit:&thinsp;&thinsp;&thinsp;&thinsp;&thinsp;&thinsp;&thinsp;&thinsp;&thinsp;&thinsp;&thinsp;&thinsp;&thinsp;&thinsp;&thinsp;&thinsp;&thinsp;[$dateRange1, $timeRange1, Wochentag(e): " . $availability1->getWeekdayNames() . "]";
+        }
+
+        return $message;
+    }
+
+    public function checkForConflictsWithExistingAvailabilities(
+        \DateTimeInterface $startDate,
+        \DateTimeInterface $endDate
+    ): ProcessList {
+        $processList = new ProcessList();
+        foreach ($this as $availability) {
+            $currentDate = clone $startDate;
+            while ($currentDate <= $endDate) {
+                if ($availability->isOpenedOnDate($currentDate)) {
+                    $conflict = $availability->getConflict();
+                    if ($conflict) {
+                        $conflictOnDay = clone $conflict;
+                        $appointmentTime = $conflictOnDay->getFirstAppointment()->getStartTime()->format('H:i');
+                        $newDate = clone $currentDate;
+                        $conflictOnDay->getFirstAppointment()->setDateTime($newDate->modify($appointmentTime));
+                        $processList->addEntity($conflictOnDay);
+                    }
+
+                    // Check for overlaps with other availabilities on this date
+                    $overlapList = $this->findOverlapsOnDate($availability, $currentDate);
+                    if ($overlapList->count()) {
+                        $processList->addList($overlapList);
+                    }
+                }
+                $currentDate = $currentDate->modify('+1day');
+            }
+        }
+
+        return $processList;
+    }
+
+    public function findOverlapsOnDate(
+        Availability $availability,
+        \DateTimeInterface $currentDate
+    ): ProcessList {
+        $processList = new ProcessList();
+
+        foreach ($this as $availabilityCompare) {
+            if ($availability === $availabilityCompare) {
+                continue;
+            }
+
+            if ($availabilityCompare->isOpenedOnDate($currentDate)) {
+                if (!$this->shouldCompareForConflicts($availability, $availabilityCompare)) {
+                    continue;
+                }
+
+                if ($this->doAvailabilitiesOverlap($availability, $availabilityCompare)) {
```

Comment:
early return:

if (!$this->doAvailabilitiesOverlap($availability, $availabilityCompare)) {
continue;
}